autonomic-administration-plugin/src/main/java/br/com/autonomiccs/autonomic/administration/plugin/AdministrationAgent.java

```diff
@@ -176,22 +183,24 @@ private ClusterVO getClusterToManageFromZone(ClusterAdministrationHeuristicAlgor
     }
 
     /**
-     * Applies a given administration algorithm onto a cluster. First it
-     * executes the {@link #processCluster(ClusterVO, ClusterAdministrationHeuristicAlgorithm)} method;
-     * then, it sets the cluster 'administration_status' to {@link ClusterAdministrationStatus#Done} and
+     * It applies a given administration algorithm onto a cluster. First it
+     * executes the {@link #processCluster(ClusterVO, ClusterAdministrationHeuristicAlgorithm)}
+     * method;
+     * then, it sets the cluster 'administration_status' to {@link ClusterAdministrationStatus#Done}
+     * and
      * 'last_administration' to the current date at the 'cloud.cluster' table.
      */
-    private void workOnCluster(ClusterVO cluster, ClusterAdministrationHeuristicAlgorithm administrationAlgorithm) {
+    public void workOnCluster(ClusterVO cluster, ClusterAdministrationHeuristicAlgorithm administrationAlgorithm) {
         long clusterId = cluster.getId();
-        logger.debug(String.format("Managing cluster[%d].", clusterId));
+        logger.debug(String.format("Managing cluster [id=%d].", clusterId));
```

Comment:
Thanks, It should be protectet.

The clusters could be marked as in progress by both ways, it is just a matter of design. Current approach marks the cluster when it is detected as available to be processed. Could be marked after the search, at the "br.com.autonomiccs.autonomic.administration.plugin.AdministrationAgent.receiveClusterToBeManaged(Long)".